### ISSUE TITLE
Access denied (403) for font CSS files loaded via Kit

### ISSUE THREAD

**Issue by ryanc-me** [FortAwesome/Font-Awesome#17346]
**Describe the bug**
At some point in the last ~1-2 weeks, the icons on my website stopped loading. After inspecting browser logs, I can see that the font CSS file URLs are returning 403 Access Denied response codes.

The browser log shows:
```
my-fa-kit.js:2 GET https://ka-f.fontawesome.com/releases/v5.14.0/css/free-v4-shims.min.css 403
my-fa-kit.js:2 Font Awesome Kit: Error
my-fa-kit.js:2 GET https://ka-f.fontawesome.com/releases/v5.14.0/css/free-v4-font-face.min.css 403
my-fa-kit.js:2 GET https://ka-f.fontawesome.com/releases/v5.14.0/css/free.min.css 403
```

My FA Kit is currently targeting v5.14.0:
![chrome_wLNO93a8mo](https://user-images.githubusercontent.com/762418/97816748-26030300-1cfc-11eb-9ba1-16251b2c2820.png)

I also tried updating my Kit with few other versions. **v5.13.0 and v5.15.1 work correctly**, but **v5.13.1 shows a different 403 error**:
```
GET https://ka-f.fontawesome.com/releases/v5.13.1/webfonts/free-fa-solid-900.woff2 net::ERR_ABORTED 403
```

I'm located in New Zealand, and seem to be connecting with the SYD4-C2 CloudFront ingress.

**To Reproduce**
Create a FontAwesome Kit, select v5.13.1 or v5.14.0, and add the kit code to a website. Font CSS or WOFF2 URLs will fail to load with a 403 error, and the icons will not show.

Alternatively, try using `curl` to request font URLs manually:
```
curl -Is https://ka-f.fontawesome.com/releases/v5.13.0/css/free.min.css | head -n 1 # response: 200
curl -Is https://ka-f.fontawesome.com/releases/v5.13.1/css/free.min.css | head -n 1 # response: 200
curl -Is https://ka-f.fontawesome.com/releases/v5.14.0/css/free.min.css | head -n 1 # response: 403
curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css | head -n 1 # response: 200

curl -Is https://ka-f.fontawesome.com/releases/v5.13.0/webfonts/free-fa-solid-900.woff2 | head -n 1 # response: 200
curl -Is https://ka-f.fontawesome.com/releases/v5.13.1/webfonts/free-fa-solid-900.woff2 | head -n 1 # response: 403
curl -Is https://ka-f.fontawesome.com/releases/v5.14.0/webfonts/free-fa-solid-900.woff2 | head -n 1 # response: 403
curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/webfonts/free-fa-solid-900.woff2 | head -n 1 # response: 200
```

**Expected behavior**
Older versions, selected via the Kit config page, work correctly.

**Screenshots**
The FontAwesome versions available on the Kit config page:
![chrome_tT2pIfvwaT](https://user-images.githubusercontent.com/762418/97817023-a3c80e00-1cfe-11eb-9f38-c968beeea90c.png)

**Version and implementation**
Version: at least v5.13.1 and v5.14.0
Browser and version: Issue is present when using `curl`.

- [ ] SVG with JS
- [x] Web Fonts with CSS
- [ ] SVG Sprites
- [ ] On the Desktop

**Bug report checklist**
- [x] I have filled out as much of the above information as I can
- [x] I have included a test case because my odds go _way_ up that the team can fix this when I do
- [x] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by saleem-hadad** [comment]
same here!

**Comment by tagliala** [comment]
Same here (Italy)

@robmadole could you please take a look?

**Comment by ngekoding** [comment]
Same here!

**Comment by tagliala** [comment]
Please do not comment without providing additional information that could help debugging the issue (i.e.: country, this may be a region-dependent bug)

If you have this issue, please just use reactions on the opening post (👍 )

**Comment by robmadole** [comment]
Thanks for the report.

We had some caching and permission issues. I think this is resolved now. Can everyone confirm?

**Comment by ryanc-me** [comment]
@robmadole that seems to have fixed v5.14.0, thank you!

However, I am still getting a 403 error on one of the `woff2` files for v5.13.1.

Browser log:
```
GET https://ka-f.fontawesome.com/releases/v5.13.1/webfonts/free-fa-solid-900.woff2 net::ERR_ABORTED 403
```

And via curl:
```
curl -Is https://ka-f.fontawesome.com/releases/v5.13.1/webfonts/free-fa-solid-900.woff2 | head -n 1 # response: 403
```

**Comment by tagliala** [comment]
```
curl -Is https://ka-f.fontawesome.com/releases/v5.13.1/webfonts/free-fa-solid-900.woff2
HTTP/1.1 200 OK
```

Getting this from Italy

**Comment by ryanc-me** [comment]
Yup, that seems to be working for me now too!

```
curl -Is https://ka-f.fontawesome.com/releases/v5.13.1/webfonts/free-fa-solid-900.woff2 | head -n 1
HTTP/2 200
```

**Comment by ngekoding** [comment]
> Thanks for the report.
> 
> We had some caching and permission issues. I think this is resolved now. Can everyone confirm?

Thank you, everything working well now. (Indonesia)

**Comment by robmadole** [comment]
Thanks all!

**Comment by saleem-hadad** [comment]
I'm still facing these issues!
![Screen Shot 2020-11-05 at 1 19 00 AM](https://user-images.githubusercontent.com/12780866/98169174-3cc87600-1f05-11eb-8775-b7b9e1f0f2cd.png)
![Screen Shot 2020-11-05 at 1 19 18 AM](https://user-images.githubusercontent.com/12780866/98169180-3e923980-1f05-11eb-8991-92f6c88cd6d6.png)

**Comment by ryanc-me** [comment]
@saleem-hadad what happens if you load `https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css` manually in a new Chrome tab, or if you request those URL's via CURL?
```
curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css | head -n 1
curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/css/free-v4-shims.min.css | head -n 1
curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/css/free-v4-font-face.min.css | head -n 1
```

They seem to be working on my side. I wonder if the CF cache serving you is still not working correctly.

You could also try this command. It should show some more info about the cache status, CloudFront POP, etc:
```
curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css
```

**Comment by saleem-hadad** [comment]
Hi @ryanc-me thanks for the response
when I load the link directly into my browser it shows `This site can’t be reached`
![Screen Shot 2020-11-06 at 6 25 16 AM](https://user-images.githubusercontent.com/12780866/98318826-d91f7500-1ff8-11eb-9e17-f42be43cc5b1.png)
And when I run any of the above commands the output is empty :/ 
![Screen Shot 2020-11-06 at 6 26 15 AM](https://user-images.githubusercontent.com/12780866/98318870-fc4a2480-1ff8-11eb-9141-7f26cdf38c7c.png)

**Comment by saleem-hadad** [comment]
However, I enable VPN it works tho.
The issue seems to be with caching.
I live in Dubai, UAE.

**Comment by ryanc-me** [comment]
@saleem-hadad that seems to be a different error to the one we were seeing above. Could you post the output of:
 - `ping https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css`
 - `curl -Is https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css` (_without_ the `| head -n 1` part)

**Comment by saleem-hadad** [comment]
Still the output empty

**Comment by ryanc-me** [comment]
@saleem-hadad Sorry about the delay here! Are you still having issues? If so, you should open a new issue to track it separately. 🙂

**Comment by androidacy-user** [comment]
And we're baccckkk lmao. All the css files just started returning 403s again, except the initial css file.

**Comment by SHSP-Aaron-Low** [comment]
I'm using a local environment right now (WordPress 6.3.1). Also using ACF Pro (6.2.11) and the Font Awesome add-on (4.0.6). I'm encountering the issue using a kit to load webfont. Here's my kit URL: https://kit.fontawesome.com/e7bad27ec8.js

I've tried to load this in via the recommended plugin and via advanced guide (adding manual enqueue to functions.php). These aren't loading the kit URL for some reason. I've even tried to intercept and filter the script tag to add the crossorigin attribute but this does not work either. 

`curl -Is https://kit.fontawesome.com/e7bad27ec8.js` 
HTTP/2 403
date: Mon, 25 Sep 2023 23:13:31 GMT
content-length: 31
access-control-allow-headers: accept, accept-langauge, content-language, content-type, fa-kit-token
access-control-allow-methods: GET, OPTIONS
access-control-allow-origin: *
access-control-max-age: 3000
cache-control: max-age=0, private, must-revalidate
x-request-id: F4hHQrnOV7Eo3AAoO7PB
cf-cache-status: MISS
server: cloudflare
cf-ray: 80c6f54c5bcd9ab7-MIA

**Comment by androidacy-user** [comment]
> I'm using a local environment right now (WordPress 6.3.1). Also using ACF Pro (6.2.11) and the Font Awesome add-on (4.0.6). I'm encountering the issue using a kit to load webfont. Here's my kit URL: https://kit.fontawesome.com/e7bad27ec8.js
> 
> I've tried to load this in via the recommended plugin and via advanced guide (adding manual enqueue to functions.php). These aren't loading the kit URL for some reason. I've even tried to intercept and filter the script tag to add the crossorigin attribute but this does not work either. 
> 
> `curl -Is https://kit.fontawesome.com/e7bad27ec8.js` 
> HTTP/2 403
> date: Mon, 25 Sep 2023 23:13:31 GMT
> content-length: 31
> access-control-allow-headers: accept, accept-langauge, content-language, content-type, fa-kit-token
> access-control-allow-methods: GET, OPTIONS
> access-control-allow-origin: *
> access-control-max-age: 3000
> cache-control: max-age=0, private, must-revalidate
> x-request-id: F4hHQrnOV7Eo3AAoO7PB
> cf-cache-status: MISS
> server: cloudflare
> cf-ray: 80c6f54c5bcd9ab7-MIA

This initial request is expected to 403 without a valid referrer. However, for us, the files loaded BY the kit where getting the 403, not the kit itself

**Comment by maiorano84** [comment]
We have a similar issue here, except with the actual Javascript file. The domain is whitelisted under the Kit settings, but we're still getting a 403 back from the CDN.

What's interesting is that only one of the whitelisted domains currently works, which is a local domain our developers use. The Staging domain continues to yield a 403 despite multiple cache clears both server-side, client-side, and in Incognito.

**Comment by rgbinnovation** [comment]
Currently experiencing this. Work fine in my localhost, but in production it fails. Note that it used to work fine. Anyone have a. fix?

**Comment by tagliala** [comment]
@robmadole do we have a new temporary problem?

**Comment by rgbinnovation** [comment]
There is no more problem, I was a moron. Had not added www.xxxxx.com to the whitelist, only xxxxx.comSent from my iPhone. Please excuse brief replies.On Dec 2, 2024, at 23:04, Geremia Taglialatela ***@***.***> wrote:﻿
@robmadole do we have a new temporary problem?

**Comment by tagliala** [comment]
Thanks for the heads-up, glad you solved

**Comment by maiorano84** [comment]
Just wanted to follow up here with some information that we discovered and figured it might help someone in the future:

Our server was sending a `Referrer-Policy` header with a `"no-referrer"` value. This was preventing the `Origin` header from being sent, which FontAwesome relies on for its whitelisting. The default policy is `strict-origin-when-cross-origin`, so we were able to simply drop this header, but if you prefer to set it explicitly then that would be the baseline value to use.
